boskos/cmd/janitor/cleaners.go

```diff
@@ -0,0 +1,11 @@
+package cleaners
+
+import (
+	"google.golang.org/api/compute/v1"
+)
+
+type cleaners interface {
+	List() ([]*compute.Instance, error)
```

Comment:
the signatures of all of these functions are tied to the compute API, so this really isn't a generic interface.

I'm almost wondering if we should be trying to reuse the existing interfaces from the aws-janitor (e.g. [this](https://github.com/kubernetes/test-infra/blob/4e8ce0af1661e4fe791444fe940384a11086235b/boskos/aws-janitor/resources/list.go#L23-L32)), possibly with some refactoring, rather than trying to start fresh.